### PR TITLE
Timestamp add

### PR DESCRIPTION
**Pull request by Dylan-Health-Union**
## What is this? 
Adding timeadd to xdb to go with dateadd

## What changes in this PR? 
* Added timeadd macro
* Added tests for the timeadd macro

## How does this impact our users?
* Users will now be able to use timeadd

## PR Quality
- [ ] does `docker-compose exec testxdb test` run successfully?
- [ ] does `docker-compose exec testxdb docs` build docs without errors?
- [ ] does `docker-compose exec testxdb coverage` pass coverage standards?
- [ ] did you leave the codebase better than you found it? 

@Health-Union/hu-data make sure to include a version tag in the merge commit!

### CODE REVIEW

File: macros/timeadd.sql [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{%- macro timeadd(part, amount_to_add, value) -%}
+    {# adds `amount_to_add` `part`s to `value`. so adding one hour to Jan 1 2020 01:00:00 would be timeadd('hour',1,'2020-01-01 01:00:00'). 
+       NOTE: dateadd only manipulates time values. for date additions see [dateadd](#dateadd)
```
Comment (norton120) [inline macros/timeadd.sql:3]: think this should be "timeadd only manipulates time values"

File: macros/timeadd.sql [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+       ARGS:
+         - part (string) one of 'second','minute','hour'.  
+         - amount_to_add (int) number of `part` units to add to `value`. Negative subtracts.
+         - value (string) the date time string or column to add to.
+       RETURNS: a date time value with the amount added.
+    #}
+    {%- set part = part |lower -%}
+    {%if part not in ('second','minute','hour') %}
+        {% if part in ('day','week','month','year') %}
+            {{exceptions.raise_compiler_error("date component passed to macro `timeadd()`. Did you want `dateadd()`?")}}
```
Comment (norton120) [inline macros/timeadd.sql:13]: this is beautiful

Comment (Dylan-Health-Union) [reply macros/timeadd.sql:13]: Do I point out that I stole this directly from dateadd which @norton120  wrote, or just take the compliment?

Comment (norton120) [reply macros/timeadd.sql:13]: I knew I liked that style..

**Pull request merged**